web/src/stats/stats.ts

```diff
@@ -298,22 +298,20 @@ function populate_messages_sent_over_time(data) {
     }
 
     const start_dates = data.end_times.map(
-        (timestamp) =>
+        (timestamp: number) =>
             // data.end_times are the ends of hour long intervals.
             new Date(timestamp * 1000 - 60 * 60 * 1000),
     );
 
     function aggregate_data(aggregation) {
-        let start;
-        let is_boundary;
+        let start = new Date();
+        let is_boundary: (date: Date) => boolean = () => false;
         if (aggregation === "day") {
             start = floor_to_local_day(start_dates[0]);
-            is_boundary = function (date) {
-                return date.getHours() === 0;
-            };
+            is_boundary = (date) => date.getHours() === 0;
         } else if (aggregation === "week") {
             start = floor_to_local_week(start_dates[0]);
```

Comment:
Maybe we should change this from `else if` to `else`, and add `assert(aggregation === "week")` here.
Setting defaults for `start` and `is_boundary` suppresses potential programming errors.